vm-deployment/vm.bicep

```
@@ -0,0 +1,218 @@
+  'StandardSSD_ZRS'
+  'Standard_LRS'
+  'UltraSSD_LRS'
+])
+param diskSku string = 'Premium_LRS'
+
+@description('The priority of the Virtual Machine.')
+@allowed([
+  'Regular'
+  'Low'
+  'Spot'
+])
+param vmPriority string = 'Regular'
+
+
+
+// Networking security group
+resource networkSecurityGroup 'Microsoft.Network/networkSecurityGroups@2023-11-01' = {
+  name: networkSecurityGroupName
+  location: location
+  properties: {
+    securityRules: [for (port, portIndex) in allowedTcpPorts: {
+        name: 'AllowTcp${port}'
+        properties: {
+          protocol: 'TCP'
+          destinationPortRange: '${port}'
+          access: 'Allow'
+          priority: 300 + (portIndex * 20)
+          direction: 'Inbound'
+        }
+      }]
+    ddosSettings: {
+      protectionMode: 'VirtualNetworkInherited'
+    }
+  }
+}
+
+// Public IP address
+resource publicIPAddressResource 'Microsoft.Network/publicIPAddresses@2023-11-01' = {
+  name: publicIPAddressName
+  location: location
+  sku: {
+    name: 'Standard'
+    tier: 'Regional'
+  }
+  properties: {
+    publicIPAddressVersion: 'IPv4'
+    publicIPAllocationMethod: 'Static'
+    ddosSettings: {
+      protectionMode: 'VirtualNetworkInherited'
+    }
+  }
+}
+
+
+// Virtual network
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2023-11-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: 'default'
+        properties: {
+          addressPrefix: subnetAddressPrefix
+          privateEndpointNetworkPolicies: 'Disabled'
+          privateLinkServiceNetworkPolicies: 'Enabled'
+        }
+      }
+    ]
+  }
+}
+
+
+
+// Virtual machine
+resource virtualMachine 'Microsoft.Compute/virtualMachines@2024-03-01' = {
+  name: virtualMachineName
+  location: location
+  properties: {
+    hardwareProfile: {
+      vmSize: vmSize
+    }
+    priority: vmPriority
+    storageProfile: {
+      imageReference: {
+        publisher: 'Canonical'
+        offer: '0001-com-ubuntu-server-jammy'
+        sku: '22_04-lts-gen2'
+        version: 'latest'
+      }
+      osDisk: {
+        osType: 'Linux'
+        name: '${virtualMachineName}_OsDisk_1'
+        createOption: 'FromImage'
+        managedDisk: {
+          storageAccountType: diskSku
+        }
+        diskSizeGB: diskSizeGB
+      }
+    }
+    osProfile: {
+      computerName: virtualMachineName
+      adminUsername: adminUsername
+      linuxConfiguration: {
+        disablePasswordAuthentication: true
+        ssh: {
+          publicKeys: [
+            {
+              path: '/home/${adminUsername}/.ssh/authorized_keys'
+              keyData: sshPublicKey
+            }
+          ]
+        }
+      }
+    }
+    networkProfile: {
+      networkInterfaces: [
+        {
+          id: networkInterfaceResource.id
+        }
+      ]
+    }
+  }
+}
+// Network interface
```

```suggestion
}

// Network interface
```
Missing newline between two resources?